### PR TITLE
Adds MathHelper.LerpPrecise

### PR DESCRIPTION
**Pull request by hede**
Following discussion on #4204, this PR implements the proposed MathHelper.LerpPrecise.

I've also fixed a few apparent typos in the tests, added an XNA compat. test for current Lerp behaviour and added the relevant tests for LerpPrecise.

### CODE REVIEW

File: MonoGame.Framework/MathHelper.cs [hunk 1]
```diff
@@ -159,19 +159,46 @@ public static float Hermite(float value1, float tangent1, float value2, float ta
...
         /// </summary>
         /// <param name="value1">Source value.</param>
-        /// <param name="value2">Source value.</param>
+        /// <param name="value2">Destination value.</param>
         /// <param name="amount">Value between 0 and 1 indicating the weight of value2.</param>
         /// <returns>Interpolated value.</returns> 
         /// <remarks>This method performs the linear interpolation based on the following formula.
         /// <c>value1 + (value2 - value1) * amount</c>
         /// Passing amount a value of 0 will cause value1 to be returned, a value of 1 will cause value2 to be returned.
+        /// <seealso cref="MathHelper.LerpPrecise"/> For a less efficient version with more precision around edge cases
```
Comment (tomspilman) [inline MonoGame.Framework/MathHelper.cs]: Missing a period at the end of the sentence.

File: MonoGame.Framework/MathHelper.cs [hunk 2]
```diff
@@ -159,19 +159,46 @@ public static float Hermite(float value1, float tangent1, float value2, float ta
...
-        }
-
-	/// <summary>
+        }
+
+
+        /// <summary>
+        /// Linearly interpolates between two values.
+        /// This method is a less efficient, more precise version of <see cref="MathHelper.Lerp"/>
+        /// See remarks for more info
```
Comment (tomspilman) [inline MonoGame.Framework/MathHelper.cs]: Missing periods at the end of the last two sentances.

File: MonoGame.Framework/MathHelper.cs [hunk 3]
```diff
@@ -159,19 +159,46 @@ public static float Hermite(float value1, float tangent1, float value2, float ta
...
+        /// <param name="value2">Destination value.</param>
+        /// <param name="amount">Value between 0 and 1 indicating the weight of value2.</param>
+        /// <returns>Interpolated value.</returns> 
+        /// <remarks>This method performs the linear interpolation based on the following formula.
+        /// <c> ((1 - amount) * value1) + (value2 * amount) </c>
+        /// Passing amount a value of 0 will cause value1 to be returned, a value of 1 will cause value2 to be returned.
+        /// This method does not have the floating point precision issue that <see cref="MathHelper.Lerp"/> has.
+        /// i.e. If there is a big gap between value1 and value2 in magnitude (e.g. value1=10000000000000000, value2=1)
+        /// Right at the edge of the interpolation range (amount=1), <see cref="MathHelper.Lerp"/> will return 0 (whereas it should return 1)
+        /// This also holds for value1=10^17, value2=10; value1=10^18,value2=10^2... so on
```
Comment (tomspilman) [inline MonoGame.Framework/MathHelper.cs]: Missing period.

**Comment by tomspilman** [comment]
Looks good... just a few doc fixes are needed.

**Comment by hede** [comment]
Should be ok now.
Sorry for the sloppy grammar.

**Comment by tomspilman** [comment]
Works for me... thanks @hede .  Merging!

**Comment by tomspilman** [comment]
If you want to apply some of this to `Vector2/3/4` that would be good.

**Pull request merged**